charts/skypilot/templates/api-deployment.yaml

```diff
@@ -52,19 +58,20 @@ spec:
           # Any local changes to the config.yaml file will be overwritten by the contents of the configmap.
           cp /tmp/config.yaml /root/.sky/config.yaml
           {{- end }}
-          sky api start --deploy && tail -f /root/.sky/api_server/server.log
+          exec python -m sky.server.server --deploy --host 0.0.0.0 --port 46580
         ports:
         - containerPort: 46580
         livenessProbe:
           httpGet:
             path: /api/health
             port: 46580
-          periodSeconds: 30
+          periodSeconds: 10
         readinessProbe:
           httpGet:
             path: /api/health
             port: 46580
-          periodSeconds: 30
+          # Accelerate startup and failover
+          periodSeconds: 10
```

Comment:
Probably should bump up `failureThreshold` to 5 or 6 to give upto a minute for the API server to start (which can take depending on the preDeployHook).